cmd/controller/main.go

```diff
@@ -134,18 +135,19 @@ func main() {
 			logger.Fatalf("failed to wait for cache at index %v to sync", i)
 		}
 	}
-
+	var wg sync.WaitGroup
```

Comment:
We could use https://godoc.org/golang.org/x/sync/errgroup here too, and just have each `g.Go` call `ctrlr.Run` and return its error.

```
var g errgroup.Group
for _, c := range controllers {
  c := c // capture c for the goroutine
  g.Go(func() error {
    return c.Run(...)
  })
}
if err := g.Wait(); err != nil {
  logger.Fatalf("...", err)
}
```

Reply:
I have never used `errgroup` before. That looks great